Source/Core/Core/AchievementManager.cpp

```diff
@@ -830,100 +327,106 @@ const AchievementManager::BadgeStatus& AchievementManager::GetGameBadge() const
   return m_game_badge;
 }
 
-const AchievementManager::UnlockStatus&
-AchievementManager::GetUnlockStatus(AchievementId achievement_id) const
+const AchievementManager::BadgeStatus& AchievementManager::GetAchievementBadge(AchievementId id,
+                                                                               bool locked) const
 {
-  return m_unlock_map.at(achievement_id);
+  if (locked)
+    return AchievementManager::GetInstance().m_locked_badges[id];
+  return AchievementManager::GetInstance().m_unlocked_badges[id];
 }
 
-AchievementManager::ResponseType
-AchievementManager::GetAchievementProgress(AchievementId achievement_id, u32* value, u32* target)
+const AchievementManager::LeaderboardStatus*
+AchievementManager::GetLeaderboardInfo(AchievementManager::AchievementId leaderboard_id)
 {
-  if (!IsGameLoaded())
-  {
-    ERROR_LOG_FMT(
-        ACHIEVEMENTS,
-        "Attempted to request measured data for achievement ID {} when no game is running.",
-        achievement_id);
-    return ResponseType::INVALID_REQUEST;
-  }
-  int result = rc_runtime_get_achievement_measured(&m_runtime, achievement_id, value, target);
-  if (result == 0)
-  {
-    WARN_LOG_FMT(ACHIEVEMENTS, "Failed to get measured data for achievement ID {}.",
-                 achievement_id);
-    return ResponseType::MALFORMED_OBJECT;
-  }
-  return ResponseType::SUCCESS;
+  if (m_leaderboard_map.count(leaderboard_id) < 1)
+    return nullptr;
+  if (m_leaderboard_map.at(leaderboard_id).entries.size() == 0)
+    FetchBoardInfo(leaderboard_id);
+  return &m_leaderboard_map.at(leaderboard_id);
 }
 
-const std::unordered_map<AchievementManager::AchievementId, AchievementManager::LeaderboardStatus>&
-AchievementManager::GetLeaderboardsInfo() const
+AchievementManager::RichPresence AchievementManager::GetRichPresence() const
 {
-  return m_leaderboard_map;
+  return m_rich_presence;
 }
 
-AchievementManager::RichPresence AchievementManager::GetRichPresence() const
+const AchievementManager::NamedIconMap& AchievementManager::GetChallengeIcons() const
 {
-  std::lock_guard lg{m_lock};
-  return m_rich_presence;
+  return m_active_challenges;
 }
 
-void AchievementManager::SetDisabled(bool disable)
+std::vector<std::string> AchievementManager::GetActiveLeaderboards() const
 {
-  bool previously_disabled;
+  std::vector<std::string> display_values;
+  for (u32 ix = 0; ix < MAX_DISPLAYED_LBOARDS && ix < m_active_leaderboards.size(); ix++)
   {
-    std::lock_guard lg{m_lock};
-    previously_disabled = m_disabled;
-    m_disabled = disable;
-    if (disable && m_is_game_loaded)
-      CloseGame();
+    display_values.push_back(std::string(m_active_leaderboards[ix].display));
```

Comment:
Do these need to be long-lived copies?  Or can you have this function return a `std::vector<std::string_view>`?

Reply:
These need to be copies if they exit the scope of the lock.